File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        String temp = "";
        while(n>0){
            temp =  n%3 +temp;
            n /= 3;
        }
        int answer =0;
        for(int x =0 ; x<temp.length();x++){
            String y ="";
            y+=temp.charAt(x);
            int b = Integer.parseInt(y);
   
            if(b!=0){
                if(x==0){
                    answer+= b;
                }else{
                    answer+= Math.pow(3, x) *b;
                }
            }
        }
   
        return answer;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.js

function solution(n) {
    let temp = "";
    let answer =0;
    while(n>0){
        temp =  n%3 +temp;
        n = parseInt(n /3);
    }
    for( let a in temp){
        answer += temp[a] * Math.pow(3,a);
    }
    return answer;
}

/*
function solution(n) {
    let answer = 0;
    let num = n.toString(3);
    for( let a in num){
         answer += num[a] * Math.pow(3,a);
    }
    return answer;
}
*/
